### PR TITLE
Misc remote control updates

### PR DESCRIPTION
**Pull request by jeromelaban**
GitHub Issue (If applicable): https://github.com/unoplatform/uno/issues/10478

## PR Type

What kind of change does this PR introduce?
- Bugfix

## What is the new behavior?

- Enables NRT for remote control assemblies
- Fix invalid deserialization for changed types
- Run metadata updates on Dispatcher

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI.RemoteControl/HotReload/ClientHotReloadProcessor.Xaml.cs [hunk 1]
```diff
@@ -44,61 +48,86 @@ private void ReloadFile(FileReload fileReload)
...
+                    targetDictionary.CopyFrom(replacementDictionary);
+                    Application.Current.UpdateResourceBindingsForHotReload();
+                }
+            }
+            catch (Exception e)
+            {
+                if (this.Log().IsEnabled(LogLevel.Error))
+                {
+                    this.Log().LogError($"Failed reloading changed file [{filePath}]", e);
+                }
```
Comment (spouliot) [inline src/Uno.UI.RemoteControl/HotReload/ClientHotReloadProcessor.Xaml.cs:119]: ^ spaces, not tabs, are being used for that block

File: src/Uno.UI.RemoteControl/HotReload/ClientHotReloadProcessor.Xaml.cs [hunk 2]
```diff
@@ -44,61 +48,86 @@ private void ReloadFile(FileReload fileReload)
...
+                    new HotReload.Messages.XamlLoadError(
+                        filePath: filePath,
+                        exceptionType: e.GetType().ToString(),
+                        message: e.Message,
+                        stackTrace: e.StackTrace));
+            }
+        }
 
-		private static IEnumerable<UIElement> EnumerateInstances(object instance, Func<FrameworkElement, bool> predicate)
+        private static IEnumerable<UIElement> EnumerateInstances(object instance, Func<FrameworkElement, bool> predicate)
```
Comment (spouliot) [inline src/Uno.UI.RemoteControl/HotReload/ClientHotReloadProcessor.Xaml.cs:130]: spaces

File: src/Uno.UI.RemoteControl/HotReload/ClientHotReloadProcessor.MetadataUpdate.cs [hunk 3]
```diff
@@ -85,24 +90,34 @@ private string[] GetMetadataUpdateCapabilities()
...
+			if (assemblyDeltaReload.IsValid())
 			{
-				this.Log().Trace($"Applying IL Delta after {assemblyDeltaReload.FilePath}, Guid:{assemblyDeltaReload.ModuleId}");
-			}
+				if (this.Log().IsEnabled(LogLevel.Trace))
+				{
+					this.Log().Trace($"Applying IL Delta after {assemblyDeltaReload.FilePath}, Guid:{assemblyDeltaReload.ModuleId}");
+				}
+
+				var changedTypesStreams = new MemoryStream(Convert.FromBase64String(assemblyDeltaReload.UpdatedTypes));
```
Comment (spouliot) [inline src/Uno.UI.RemoteControl/HotReload/ClientHotReloadProcessor.MetadataUpdate.cs:100]: ah, I missed that in the previous PR

File: src/Uno.UI.RemoteControl/HotReload/ClientHotReloadProcessor.MetadataUpdate.cs [hunk 4]
```diff
@@ -122,6 +137,121 @@ static int[] ReadIntArray(BinaryReader binaryReader)
...
+
+                return;
+            }
+
+            if (_log.IsEnabled(LogLevel.Debug))
+            {
+                _log.LogDebug($"Processing last changed file [{lastUpdated}]");
+            }
+
+            var uri = new Uri("file:///" + lastUpdated.Replace("\\", "/"));
```
Comment (spouliot) [inline src/Uno.UI.RemoteControl/HotReload/ClientHotReloadProcessor.MetadataUpdate.cs:196]: `Replace('\', '/')`

File: src/Uno.UI.RemoteControl/HotReload/ClientHotReloadProcessor.MetadataUpdate.cs [hunk 5]
```diff
@@ -122,6 +137,121 @@ static int[] ReadIntArray(BinaryReader binaryReader)
...
+
+                    switch (instance)
+                    {
+#if __IOS__
+						case UserControl userControl:
+							if (newInstance is UIKit.UIView newContent)
+                            {
+                                SwapViews(userControl, newContent);
+							}
+							break;
```
Comment (spouliot) [inline src/Uno.UI.RemoteControl/HotReload/ClientHotReloadProcessor.MetadataUpdate.cs:224]: ^ alignment is incorrect. Seems spaces and tabs are mixed here

File: src/Uno.UI.RemoteControl/HotReload/ClientHotReloadProcessor.Xaml.cs [hunk 6]
```diff
@@ -44,61 +48,89 @@ private void ReloadFile(FileReload fileReload)
...
+        private async Task ReloadWithFileAndContent(string filePath, string fileContent)
+        {
+            try
+            {
+                if (this.Log().IsEnabled(LogLevel.Debug))
+                {
+                    this.Log().LogDebug($"Reloading changed file [{filePath}]");
+                }
+
+                var uri = new Uri("file:///" + filePath.Replace("\\", "/"));
```
Comment (spouliot) [inline src/Uno.UI.RemoteControl/HotReload/ClientHotReloadProcessor.Xaml.cs:85]: same, replace `char`s not `string`s

File: src/Uno.UI.RemoteControl/HotReload/ClientHotReloadProcessor.Xaml.cs [hunk 7]
```diff
@@ -44,61 +48,89 @@ private void ReloadFile(FileReload fileReload)
...
+                foreach (var instance in EnumerateInstances(Window.Current.Content, i => uri.OriginalString == i.BaseUri?.OriginalString))
+                {
+                    switch (instance)
+                    {
+#if __IOS__
+						case UserControl userControl:
+							if (XamlReader.LoadUsingXClass(fileContent) is UIKit.UIView newInstance)
+							{
+								SwapViews(userControl, newInstance);
+							}
```
Comment (spouliot) [inline src/Uno.UI.RemoteControl/HotReload/ClientHotReloadProcessor.Xaml.cs:98]: same, mixed spaces and tabs

File: src/Uno.UI.RemoteControl/HotReload/ClientHotReloadProcessor.Xaml.cs (hunk 1 above)
Comment (jeromelaban) [reply src/Uno.UI.RemoteControl/HotReload/ClientHotReloadProcessor.Xaml.cs:122]: Conditional blocks are fun for that :)

**Review by spouliot — APPROVED** [review]

**Pull request merged**